### DISCUSSION TITLE
Home View Templating

### DISCUSSION THREAD

**Discussion by phyber** [abdolence/slack-morphism-rust#204] [Q&A] [answered]
Hi,

I'm wondering if any templating for the Home View is currently implemented in the crate. The templating for `SlackMessageContent` was simple and easy to use thanks to the Block Kit documentation.

If templating for the Home View is available, would it be possible to add an example of a basic Home View to the Block Kit documentation or the repository? I'm struggling to work out which things the crate wants from me to template a Home View.

Thanks,

**Comment by abdolence** [comment, marked as answer]
Hey, the `client` example already had Slack Block kit home:
https://github.com/abdolence/slack-morphism-rust/blob/5b4eedddbf7a27ebb6eee47417436083e7d487a7/examples/client.rs#L144

Are you looking for something like this?

**Comment by phyber** [comment]
Oh, excellent, thanks!

I’m not sure how I missed that one (probably because I was grepping for `HomeView`). My guess as to what I needed was very close too.

Thanks again, and thanks for the crate 👍

**Comment by abdolence** [comment]
My pleasure.

Left you more advanced example here:
https://github.com/abdolence/slack-morphism-rust/blob/15c2e38fa9f85220c4bef37487da60770285378e/examples/socket_mode.rs#L63

So, you can see how to work with templates and render dynamic stuff in it.
Probably can be optimised even more, but just to give an idea.
